### PR TITLE
Change constructor with default params

### PR DESCRIPTION
**Pull request by irux**

### CODE REVIEW

File: faust_s3_backed_serializer/s3_serializer.py [hunk 1]
```diff
@@ -38,11 +38,11 @@ class S3BackedSerializer(Codec):
 
     """
 
-
     VALUE_PREFIX = "values"
     KEY_PREFIX = "keys"
 
-    def __init__(self, output_topic: str, base_path: str, region_name: str, s3_credentials: Dict[str, str] = None,
+    def __init__(self, output_topic: str = None, base_path: str = None, region_name: str = None,
```
Comment (philipp94831) [inline faust_s3_backed_serializer/s3_serializer.py]: If None is allowed, then type must be Optional[str]

File: faust_s3_backed_serializer/s3_serializer.py [hunk 2]
```diff
@@ -79,20 +79,26 @@ def __init__(self, output_topic: str, base_path: str, region_name: str, s3_crede
         ------
         ValueError
             If 'base_path' param is empty or not given
+            If 'output_topic' params is empty or not given
```
Comment (philipp94831) [inline faust_s3_backed_serializer/s3_serializer.py]: upon write

File: faust_s3_backed_serializer/s3_serializer.py [hunk 3]
```diff
@@ -79,20 +79,26 @@ def __init__(self, output_topic: str, base_path: str, region_name: str, s3_crede
...
 
         super().__init__(output_topic=output_topic, base_path=base_path, max_size=max_size,
                          s3_credentials=s3_credentials, region_name=region_name,
                          **kwargs)
 
+        self._base_path = base_path
+        self._max_size = max_size
+        self._is_key = is_key
+        self._output_topic = output_topic
+        self._verify_params()
```
Comment (philipp94831) [inline faust_s3_backed_serializer/s3_serializer.py]: Only check this if dumps is called and message is too large

Comment (irux) [reply faust_s3_backed_serializer/s3_serializer.py]: what about now?

File: tests/s3_serializer_test.py [hunk 4]
```diff
@@ -108,7 +108,7 @@ def test_dumps_from_s3_serde_min_size_value_error(s3_serializer_factory, s3_clie
         with pytest.raises(S3UploadException) as e:
             data_bytes = b"this is my data"
             s3_uri = s3_serializer_min_size.dumps(data_bytes)
-            assert e.value == "The S3 client couldn't upload the data to the s3 bucket"
+        assert str(e.value) == "The S3 client couldn't upload the data to the s3 bucket"
```
Comment (philipp94831) [inline tests/s3_serializer_test.py]: is it correct that you removed the indentation?

File: faust_s3_backed_serializer/s3_serializer.py (hunk 3 above)
Comment (philipp94831) [reply faust_s3_backed_serializer/s3_serializer.py]: Looks good.Please update the constructor doc

File: faust_s3_backed_serializer/s3_serializer.py [hunk 5]
```diff
@@ -38,13 +38,14 @@ class S3BackedSerializer(Codec):
...
     KEY_PREFIX = "keys"
 
-    def __init__(self, output_topic: str, base_path: str, region_name: str, s3_credentials: Dict[str, str] = None,
-                 max_size: int = int(1e6),
-                 is_key=False,
+    def __init__(self, output_topic: Optional[str] = None, base_path: Optional[str] = None,
+                 region_name: Optional[str] = None,
+                 s3_credentials: Optional[Dict[str, str]] = None,
+                 max_size: Optional[int] = int(1e6),
+                 is_key: Optional[bool] = False,
```
Comment (philipp94831) [inline faust_s3_backed_serializer/s3_serializer.py]: That should not be Optional :)

File: faust_s3_backed_serializer/s3_serializer.py [hunk 6]
```diff
@@ -38,13 +38,14 @@ class S3BackedSerializer(Codec):
...
     VALUE_PREFIX = "values"
     KEY_PREFIX = "keys"
 
-    def __init__(self, output_topic: str, base_path: str, region_name: str, s3_credentials: Dict[str, str] = None,
-                 max_size: int = int(1e6),
-                 is_key=False,
+    def __init__(self, output_topic: Optional[str] = None, base_path: Optional[str] = None,
+                 region_name: Optional[str] = None,
+                 s3_credentials: Optional[Dict[str, str]] = None,
+                 max_size: Optional[int] = int(1e6),
```
Comment (philipp94831) [inline faust_s3_backed_serializer/s3_serializer.py]: No Optional

File: tests/s3_serializer_test.py (hunk 4 above)
Comment (irux) [reply tests/s3_serializer_test.py]: yeap, it was a mistake before! the identation should not be there

File: faust_s3_backed_serializer/s3_serializer.py (hunk 6 above)
Comment (irux) [reply faust_s3_backed_serializer/s3_serializer.py]: Like this?

File: faust_s3_backed_serializer/s3_serializer.py (hunk 3 above)
Comment (irux) [reply faust_s3_backed_serializer/s3_serializer.py]: the types right ?

Comment (philipp94831) [reply faust_s3_backed_serializer/s3_serializer.py]: And the Exception part :)

Comment (irux) [reply faust_s3_backed_serializer/s3_serializer.py]: like this ?

**Pull request merged**